tutorials/large-messages/index.mdx

```diff
@@ -0,0 +1,370 @@
+
+3. Create a `terraform.tfvars` file and add your Scaleway access key and access key id to it.
+    ```terraform
+    scw_access_key_id     = "YOUR_SCW_SECRET_KEY_ID"
+    scw_secret_access_key = "YOUR_SCW_SECRET_KEY"
+    ```
+
+4. Continuing in the `main.tf` file, add the following Terraform code to create an Object Storage bucket that will be used for storing your images.
+    ```terraform
+    resource "random_id" "bucket" {
+      byte_length = 8
+    }
+
+    resource "scaleway_object_bucket" "large_messages" {
+      name = "large-messages-${random_id.bucket.hex}"
+    }
+
+    resource "scaleway_object_bucket_acl" "large_messages" {
+      bucket = scaleway_object_bucket.large_messages.id
+      acl    = "private"
+    }
+
+    output "bucket_name" {
+      value       = scaleway_object_bucket.large_messages.name
+      description = "Bucket name to use with the producer script"
+    }
+    ```
+    In this code, the resource `random_id.bucket` generates a random ID, which is then passed to the object bucket to ensure its uniqueness. Additionally, a `scaleway_object_bucket_acl` ACL is applied to the bucket, setting it to private and outputting the bucket name for use in your producer.
+
+5. Add these resources to create a NATS account and your NATS credential file:
+    ```terraform
+    resource "scaleway_mnq_nats_account" "large_messages" {
+      name = "nats-acc-large-messages"
+    }
+
+    resource "scaleway_mnq_nats_credentials" "large_messages" {
+      name       = "nats-large-messages-creds"
+      account_id = scaleway_mnq_nats_account.large_messages.id
+    }
+
+    resource "local_file" "nats_credential" {
+      content         = scaleway_mnq_nats_credentials.large_messages.file
+      filename        = "large-messages.creds"
+      file_permission = 644
+    }
+
+    output "nats_url" {
+      value       = scaleway_mnq_nats_account.large_messages.endpoint
+      description = "NATS url to use with the producer script"
+    }
+    ```
+    We also output the NATS server url.
+
+6. You can run `terraform init` and `terraform apply` to create the resources in your Scaleway account. 
+
+## Create the producer
+
+As mentioned earlier, the producer will be implemented as a straightforward shell script.
+
+1. Create a file named `upload_img.sh`. Inside it, define two variables using Terraform commands to retrieve the bucket name and the NATS URL.
+    ```bash
+    SCW_BUCKET="$(terraform output bucket_name)"
+    SCW_NATS_URL="$(terraform output nats_url)"
+    ```
+
+2. Configure the NATS CLI with the NATS URL and the credentials file.
+    ```bash
+    nats context save large-messages --server=$SCW_NATS_URL --creds=./large-messages.creds
+    nats context select large-messages
+    ```
+    Our script take the file path that we want to upload as first parameter.
+
+To upload the file, we will use the AWS CLI configured with the Scaleway endpoint and credentials, because the Scaleway Object storage is fully compliant with S3.
```

Comment:
```suggestion
    To upload the file, we will use the AWS CLI configured with the Scaleway endpoint and credentials, because Scaleway Object storage is fully compliant with S3.
    
```